packages/python-packages/doc-warden/warden/enforce_changelog_content.py

```diff
@@ -0,0 +1,133 @@
+    omitted_changelogs = get_omitted_files(config)
+    known_issue_paths = config.get_known_content_issues()
+    missing_changelog = []
+    empty_release_notes = []
+    changelog_results = []
+
+    for pkg in pkg_list:
+        if pkg.relative_changelog_location == '': continue
+        changelog_ext = os.path.splitext(pkg.relative_changelog_location)[1]
+        pkg_changelog = os.path.normpath(os.path.join(config.target_directory, pkg.relative_changelog_location))
+
+        if os.path.isfile(pkg_changelog) and pkg_changelog not in omitted_changelogs:
+            if changelog_ext == '.rst':
+                changelog_results.append(verify_rst_changelog(pkg_changelog, config, pkg.package_version))
+            else:
+                changelog_results.append(verify_md_changelog(pkg_changelog, config, pkg.package_version))
+
+    for changelog_tuple in changelog_results:
+        if changelog_tuple[0] in known_issue_paths:
+            continue
+        if changelog_tuple[1]['curr_pkg_version'] != changelog_tuple[1]['latest_version_entry']:
+            missing_changelog.append(changelog_tuple)
+        elif len(changelog_tuple[1]['latest_release_notes']) == 0:
+            empty_release_notes.append(changelog_tuple)
+
+    return missing_changelog, empty_release_notes
+
+# parse rst to html, check for presence of appropriate version
+def verify_rst_changelog(changelog, config, pkg_version):
+    with open(changelog, 'r', encoding="utf-8") as f:
+        changelog_content = f.read()
+    html_changelog_content = rst_to_html(changelog_content)
+    html_soup = bs4.BeautifulSoup(html_changelog_content, "html.parser")
+
+    changelog_check_result = verify_latest_rst_section(html_soup, pkg_version)
+
+    return changelog, changelog_check_result
+
+# parse md to html, check for presence of the latest version
+def verify_md_changelog(changelog, config, pkg_version):
+    if config.verbose_output:
+        print('Examining content in {}'.format(changelog))
+
+    with open(changelog, 'r', encoding="utf-8") as f:
+        changelog_content = f.read()
+    html_changelog_content = markdown2.markdown(changelog_content)
+    html_soup = bs4.BeautifulSoup(html_changelog_content, "html.parser")
+
+    changelog_check_result = verify_latest_md_section(html_soup, pkg_version)
+
+    return changelog, changelog_check_result
+
+# get details from latest changelog entry
+def verify_latest_rst_section(html_soup, pkg_version):
+    changelog_check_result = {
+        'curr_pkg_version' : pkg_version,
+        'latest_version_entry' : '',
+        'latest_release_notes' : list()
+    }
+
+    latest_entry = html_soup.find('div', id='id1')
+    if latest_entry == None: latest_entry = html_soup.find('div', id='release-history')
+
+    if latest_entry != None:
+        for entry in latest_entry:
+            if entry.name == 'h1' or entry.name == 'h2':
+                changelog_check_result['latest_version_entry'] = entry.text.split(' ')[0]
```

Comment:
Can you drop a snippet of the HTML in a comment above? Will help with ease of debugging later.